src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,123 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V> [] table;
+    private int size;
+
+    public MyHashMap() {
+        if (table == null) {
+            table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = hash(key);
+        if (table[index] != null) {
+            if (equalsKey(table[index].key, key)) {
+                table[index].value = value;
+                return;
+            }
+            if (hasNext(table[index])) {
+                Node<K, V> current = table[index];
+                while (hasNext(current)) {
+                    current = current.next;
+                    if (equalsKey(current.key, key)) {
+                        current.value = value;
+                        return;
+                    }
+                    if (!hasNext(current)) {
+                        current.next = new Node<>(key, value, null);
+                        size++;
+                        return;
+                    }
+                }
+            } else {
+                table[index].next = new Node<>(key, value, null);
+                size++;
+            }
+        } else {
+            table[index] = new Node<>(key, value, null);
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> node : table) {
+            if (node == null) {
+                continue;
```

Comment:
If you have a key, you don't need to iterate over the table. Just calculate the position in the table(by calling you "hash" method) and then check all linked nodes at this position (bucket).